lib/tail-file.js

```diff
@@ -167,8 +167,11 @@ class TailFile extends Readable {
         // First iteration - nothing has been polled yet
         this[kStartPos] = eof
       } else if (this[kInode] !== stats.ino) {
-        // File renamed/rolled
-        await this._readRemainderFromFileHandle()
+        // File renamed/rolled between polls without triggering `ENOENT`.
+        // Conditional since this *may* have already been done if `ENOENT` threw earlier.
+        if (this[kFileHandle]) {
+          await this._readRemainderFromFileHandle()
```

Comment:
I think we should also catch the error here, emit a tail error and continue, noting that we tried to read the reminder but was not possible.

Reply:
Yeah, that's fair.  The original thought here was to let it throw and be caught below which would result in a fatal `this.quit` error.  That begs the question of whether or not this is a fatal error condition, but it would seem not since it's erroring on the "old" FH.  I can add a try/catch around this specific call.

Reply:
fixed